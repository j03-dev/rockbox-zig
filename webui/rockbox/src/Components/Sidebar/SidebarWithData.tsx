import { FC } from "react";
import Sidebar from "./Sidebar";
import { controlBarState } from "../ControlBar/ControlBarState";
import { useRecoilValue } from "recoil";
import { settingsState } from "../Settings/SettingsState";

const SidebarWithData: FC<{ active: string }> = (props) => {
  const { nowPlaying } = useRecoilValue(controlBarState);
  const { enableBlur } = useRecoilValue(settingsState);
  return (
    <Sidebar {...props} cover={enableBlur ? nowPlaying?.cover : undefined} />
  );
};

export default SidebarWithData;
