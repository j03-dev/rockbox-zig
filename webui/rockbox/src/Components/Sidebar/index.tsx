import Sidebar from "./SidebarWithData";

export default Sidebar;
