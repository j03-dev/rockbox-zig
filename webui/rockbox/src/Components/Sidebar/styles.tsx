import styled from "@emotion/styled";
import { css } from "@emotion/react";
import { Link } from "react-router-dom";

export const SettingsButton = styled.button`
  display: flex;
  background-color: #f6f9fc;
  border: none;
  cursor: pointer;
  margin-top: 3px;
  height: 64px;
`;

export const Header = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`;

export const SidebarContainer = styled.div<{ cover?: string }>`
  display: flex;
  flex-direction: column;
  height: 100vh;
  width: 222px;
  background-color: #f6f9fc;
  padding: 20px;
  ${(props) =>
    props.cover &&
    css`
      background-color: #fff;
    `}
`;

export const MenuItem = styled(Link)<{ color?: string }>`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  flex-direction: row;
  padding: 10px;
  cursor: pointer;
  font-size: 14px;
  text-decoration: none;
  ${({ color }) =>
    color &&
    css`
      color: ${color};
    `}
`;
