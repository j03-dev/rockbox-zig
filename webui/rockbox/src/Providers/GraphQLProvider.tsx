import {
  ApolloClient,
  createHttpLink,
  InMemoryCache,
  ApolloProvider,
  split,
  from,
} from "@apollo/client";
import { removeTypenameFromVariables } from "@apollo/client/link/remove-typename";
import { WebSocketLink } from "@apollo/client/link/ws";
import { getMainDefinition } from "@apollo/client/utilities";
import { SubscriptionClient } from "subscriptions-transport-ws";
import { FC, ReactNode } from "react";

const uri =
  process.env.NODE_ENV === "development"
    ? import.meta.env.VITE_APP_API_URL || "http://localhost:6062/graphql"
    : `${origin}/graphql`;

const removeTypenameLink = removeTypenameFromVariables();

const httpLink = createHttpLink({
  uri,
});

const wsLink = new WebSocketLink(
  new SubscriptionClient(uri.replace("http", "ws"))
);

const splitLink = split(
  ({ query }) => {
    const definition = getMainDefinition(query);
    return (
      definition.kind === "OperationDefinition" &&
      definition.operation === "subscription"
    );
  },
  wsLink,
  from([removeTypenameLink, httpLink])
);
const link = splitLink;

const client = new ApolloClient({
  link,
  cache: new InMemoryCache(),
});

const GraphQLProvider: FC<{
  children: ReactNode;
}> = ({ children }) => {
  return <ApolloProvider client={client}>{children}</ApolloProvider>;
};

export default GraphQLProvider;
