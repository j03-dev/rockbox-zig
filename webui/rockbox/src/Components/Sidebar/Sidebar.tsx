import { FC } from "react";
import { SidebarContainer, MenuItem } from "./styles";
import { Disc } from "@styled-icons/boxicons-regular";
import { HardDrive } from "@styled-icons/feather";
import Artist from "../Icons/Artist";
import Track from "../Icons/Track";
import RockboxLogo from "../../Assets/rockbox-icon.svg";

export type SidebarProps = {
  active: string;
  cover?: string;
};

const Sidebar: FC<SidebarProps> = ({ active, cover }) => {
  return (
    <SidebarContainer cover={cover}>
      <a href="/" style={{ textDecoration: "none" }}>
        <img
          src={RockboxLogo}
          alt="Rockbox"
          style={{ width: 40, marginBottom: 20, marginLeft: 12 }}
        />
      </a>
      <MenuItem
        color={active === "albums" ? "#fe099c" : "initial"}
        to="/albums"
      >
        <Disc
          size={20}
          style={{ marginRight: 6 }}
          color={active === "albums" ? "#fe099c" : "initial"}
        />
        <div>Albums</div>
      </MenuItem>
      <MenuItem
        color={active === "artists" ? "#fe099c" : "initial"}
        to="/artists"
      >
        <Artist
          width={20}
          height={20}
          color={active === "artists" ? "#fe099c" : "initial"}
        />
        <div style={{ marginLeft: 6 }}>Artists</div>
      </MenuItem>
      <MenuItem color={active === "songs" ? "#fe099c" : "initial"} to="/tracks">
        <Track height={20} color={active === "songs" ? "#fe099c" : "initial"} />
        <div style={{ marginLeft: 6 }}>Songs</div>
      </MenuItem>
      <MenuItem color={active === "files" ? "#fe099c" : "initial"} to="/files">
        <HardDrive
          size={19}
          style={{ marginRight: 6 }}
          color={active === "files" ? "#fe099c" : "initial"}
        />
        <div>Files</div>
      </MenuItem>
    </SidebarContainer>
  );
};

export default Sidebar;
