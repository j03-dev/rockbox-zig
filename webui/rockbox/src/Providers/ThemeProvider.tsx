import { createContext, useState, FC } from "react";
import { ThemeProvider as EmotionThemeProvider } from "@emotion/react";
import {
  BaseUIDarkTheme,
  BaseUILightTheme,
  DarkTheme,
  LightTheme,
} from "../Theme";
import { BaseProvider } from "baseui";
import { ThemeProvider as MaterialThemeProvider } from "@mui/material/styles";
import { createTheme } from "@mui/material/styles";

export type Theme = "light" | "dark";

export type ThemeContextType = {
  theme: Theme;
  setTheme: (theme: Theme) => void;
};

export const ThemeContext = createContext<ThemeContextType>({
  theme: "light",
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  setTheme: (_theme: Theme) => {},
});

const muitheme = createTheme({
  cssVariables: true,
  palette: {
    primary: {
      main: "#fe099c",
    },
  },
});

export type ThemeProviderProps = {
  children: React.ReactNode;
};

const ThemeProvider: FC<ThemeProviderProps> = ({ children }) => {
  const [theme, setTheme] = useState<Theme>("light");
  return (
    <MaterialThemeProvider theme={muitheme}>
      <ThemeContext.Provider value={{ theme, setTheme }}>
        <EmotionThemeProvider theme={theme === "dark" ? DarkTheme : LightTheme}>
          <BaseProvider
            theme={theme === "dark" ? BaseUIDarkTheme : BaseUILightTheme}
          >
            {children}
          </BaseProvider>
        </EmotionThemeProvider>
      </ThemeContext.Provider>
    </MaterialThemeProvider>
  );
};

export default ThemeProvider;
