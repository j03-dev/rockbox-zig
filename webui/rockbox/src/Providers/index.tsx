import { FC } from "react";
import ThemeProvider from "./ThemeProvider";
import { Provider as StyletronProvider } from "styletron-react";
import { Client as Styletron } from "styletron-engine-atomic";
import { RecoilRoot } from "recoil";

const engine = new Styletron();

export type ProvidersProps = {
  children: React.ReactNode;
};

const Providers: FC<ProvidersProps> = ({ children }) => {
  return (
    <RecoilRoot>
      <StyletronProvider value={engine}>
        <ThemeProvider>{children}</ThemeProvider>
      </StyletronProvider>
    </RecoilRoot>
  );
};

export default Providers;
