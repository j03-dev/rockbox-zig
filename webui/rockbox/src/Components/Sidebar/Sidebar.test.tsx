import { render } from "@testing-library/react";
import Sidebar from "./Sidebar";
import Providers from "../../Providers";

describe("Sidebar", () => {
  it("should render", () => {
    const { container } = render(
      <Providers>
        <Sidebar active="albums" />
      </Providers>
    );
    expect(container).toMatchSnapshot();
  });
});
