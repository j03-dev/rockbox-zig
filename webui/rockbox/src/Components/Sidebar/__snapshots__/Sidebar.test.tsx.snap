// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sidebar > should render 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="css-1tlxqlf"
    >
      <a
        href="/"
        style="text-decoration: none;"
      >
        <img
          alt="Rockbox"
          src="/src/Assets/rockbox-icon.svg"
          style="width: 40px; margin-bottom: 20px; margin-left: 12px;"
        />
      </a>
      <a
        class="css-8v05qj"
        color="#fe099c"
        href="/albums"
      >
        <svg
          aria-hidden="true"
          class="StyledIconBase-sc-ea9ulj-0 dmvaRK"
          color="#fe099c"
          fill="currentColor"
          focusable="false"
          height="20"
          style="margin-right: 6px;"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2a10 10 0 1 0 10 10A10 10 0 0 0 12 2zm0 18a8 8 0 1 1 8-8 8 8 0 0 1-8 8z"
          />
          <path
            d="M12 8a4 4 0 1 0 4 4 4 4 0 0 0-4-4zm0 6a2 2 0 1 1 2-2 2 2 0 0 1-2 2z"
          />
        </svg>
        <div>
          Albums
        </div>
      </a>
      <a
        class="css-tujm8o"
        color="initial"
        href="/artists"
      >
        <svg
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 4.22a5.67 5.67 0 0 0-9.68 4.57l-8 9.79 3.3 3.3 9.79-8c.18 0 .36.05.55.05a5.7 5.7 0 0 0 4-9.73ZM5.74 19.86l-1.38-1.38 6.44-7.89a5.48 5.48 0 0 0 2.83 2.84Zm13.21-8.65a4.2 4.2 0 1 1 0-5.94 4.17 4.17 0 0 1 0 5.95Z"
            fill="initial"
          />
        </svg>
        <div
          style="margin-left: 6px;"
        >
          Artists
        </div>
      </a>
      <a
        class="css-tujm8o"
        color="initial"
        href="/tracks"
      >
        <svg
          height="20"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.1 4.65v11.26a3.45 3.45 0 1 0 1.5 2.84V5.85l10.2-2.36v10.62A3.45 3.45 0 1 0 21.3 17V1.61Zm-2 16a2 2 0 1 1 2-2 2 2 0 0 1-1.95 2.05Zm11.7-1.8a1.95 1.95 0 1 1 2-1.85 2 2 0 0 1-1.95 1.9Z"
            fill="initial"
          />
        </svg>
        <div
          style="margin-left: 6px;"
        >
          Songs
        </div>
      </a>
      <a
        class="css-tujm8o"
        color="initial"
        href="/files"
      >
        <svg
          aria-hidden="true"
          class="StyledIconBase-sc-ea9ulj-0 dmvaRK"
          color="initial"
          fill="none"
          focusable="false"
          height="19"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          style="margin-right: 6px;"
          viewBox="0 0 24 24"
          width="19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="22"
            x2="2"
            y1="12"
            y2="12"
          />
          <path
            d="M5.45 5.11 2 12v6a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2v-6l-3.45-6.89A2 2 0 0 0 16.76 4H7.24a2 2 0 0 0-1.79 1.11z"
          />
          <line
            x1="6"
            x2="6.01"
            y1="16"
            y2="16"
          />
          <line
            x1="10"
            x2="10.01"
            y1="16"
            y2="16"
          />
        </svg>
        <div>
          Files
        </div>
      </a>
    </div>
  </div>
  <div
    class=""
  />
</div>
`;
